react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component{
+    constructor(prop) {
+        super(prop);
+        this.state = {email: ''};
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleChange(event){
+        this.setState({
+            email: event
+        });
+    }
+
+    handleSubmit(event){
+        console.log("Email:", this.state.email.target.value);
+        event.preventDefault();
+    }
+
+
+    render(){
+        return(
+            <form onSubmit={this.handleSubmit}>
+                <label>
+                    Email
+                    <input type="text" onChange={this.handleChange}/>
```

Comment:
Here you need to control the input by setting the `value` of the input with state.

```JSX
<input type="text" onChange={this.handleChange} value={this.state.email}/>